src/Covenant/Constant.hs

```diff
@@ -39,6 +97,7 @@ data AConstant
   | AString Text
   | APair AConstant AConstant
   | AList (Vector AConstant)
+  | AData PlutusData
```

Comment:
This means, that Integer can be represented 2 ways:
`AnInteger 1` and `AData $ PlutusI 1`.
Same applies for others.

Reply:
Yes, but this is intentional. Those two mean quite different things in Plutus (and UPLC): the first is a literal integer constant, the other is specifically a `Data` encoding.

Reply:
Ah, you're right, I get it now, it actually has different semantics in this context :+1: